File: week1_C/pset1/mario2/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    char sep = ' ';
    int h;
    int s;
    int z;
    
    do 
    {
        h = get_int("Pyramid height: \n");
    }while (h < 1 || h > 8);
    
    for (int y = 1; y <= h; y++)
    {
        //y=3
        //h=5
        //..###..###
        s = h - y; 
        
        for (int t = 1; t <= s; t++)
        {
            printf("%c", sep);
        }
   
        //..
        for (z = 1; z <= y; z++)
        {
            printf("#");
        }
        
        //..###
        printf("%c%c", sep, sep);
        
        //..###..
        for (z = 1; z <= y; z++)
        {
            printf("#");
        }
        //..###..###
        printf("\n");
    }
}

File: week6_Python/pset6/mario_more/sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    # request height of the pyramid
    h = request_input()
    
    # print the pyramid
    print_pyramid(h)


# validate user input
def request_input():

    height = get_int("Enter pyramind height(1-8):\n")

    while height < 1 or height > 8:
        print("Try again!")
        height = get_int("Enter pyramind height(1-8):\n")

    return height


# print half pyramid based on height
def print_pyramid(height):
    for i in range(0, height):
        print(" "*(height-i-1), end="")
        print("#"*(i+1) + "  " + "#"*(i+1))


if __name__ == "__main__":
    main()
